src/main/java/com/codeborne/selenide/Selenide.java

```diff
@@ -101,6 +104,35 @@ public static void open(String relativeOrAbsoluteUrl, String domain, String logi
     mockModalDialogs();
   }
 
+  /**
+   *
+   * The main starting point in your tests.
+   * <p>
+   * Open browser and pass authentication using build-in proxy.
+   * <p>
+   * A common authenticationType is "Basic". See Web HTTP reference for other types.
+   * <p>
+   * {@code Configuration.fileDownload == Configuration.FileDownloadMode.PROXY;} - is required configuration.
+   *
+   * @param relativeOrAbsoluteUrl
+   * @param authenticationType
+   * @param login
+   * @param password
+   *
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Proxy-Authorization">Web HTTP reference</a>
+   * @see AuthenticationType
+   */
+  public static void open(final String relativeOrAbsoluteUrl, final AuthenticationType authenticationType,
+                          final String login, final String password) {
+    getAndCheckWebDriver();
```

Comment:
Here is too much logic for `Selenide` class. All this stuff should be in `SelenideProxyServer` and/or `navigator.open`. In this case we would not need to increase `ClassFanOutComplexity` in checkstyle.xml